test/BUILD.gn

```diff
@@ -888,7 +888,308 @@ if (!is_android) {
     public_deps = [ ":browser_tests_runner" ]
   }
 } else {  # if (!is_android) {
-  group("brave_browser_tests") {
+  test("brave_browser_tests") {
+    configs += [ "//build/config:precompiled_headers" ]
+    defines = [
+      "HAS_OUT_OF_PROC_TEST_RUNNER",
+      "CHROME_VERSION_MAJOR=" + chrome_version_major,
+    ]
+    use_default_launcher = false
+    android_manifest =
+        "${target_gen_dir}/brave_browser_tests_manifest/AndroidManifest.xml"
+    android_manifest_dep = ":brave_browser_tests_manifest"
+
+    # Skip for now because of non-public deps in //chrome/test
+    check_includes = false
+
+    deps = [
+      ":brave_browser_tests_assets",
+      ":brave_browser_tests_java",
+      ":sync_integration_test_support",
+      ":sync_integration_test_support_java",
+      ":test_support_java",
+      "//build:chromeos_buildflags",
+      "//chrome:chrome_android_core",
+      "//chrome/android:app_hooks_java",
+      "//chrome/browser/flags:flags_android",
+      "//chrome/browser/metrics:test_support",
+      "//chrome/browser/payments:browser_tests",
+      "//chrome/browser/privacy_budget:browser_tests",
+      "//chrome/browser/profiling_host:profiling_browsertests",
+      "//chrome/browser/webaudio:browser_tests",
+      "//chrome/test:browser_tests_runner",
+      "//chrome/test:test_support",
+      "//chrome/test:test_support_ui_android",
+      "//components/browser_ui/widget/android:java",
+      "//components/crash/android:crashpad_main",
+      "//components/externalauth/android:google_delegate_public_impl_java",
+      "//components/metrics/demographics:test_support",
+      "//components/page_load_metrics/browser:test_support",
+      "//components/page_load_metrics/browser/observers/ad_metrics:test_support",
+      "//components/ukm:ukm_test_helper",
+
+      # TODO(crbug.com/961849): This is needed for ShellManager which is what
+      # the ChromeBrowserTestsActivity is using to build the java UI. It's
+      # likely we want to replace it with something new that uses ChromeWindow
+      # and CompositorViewHolder instead.
+      "//content/shell:content_shell_lib",
+      "//content/test:android_test_message_pump_support",
+      "//testing/android/native_test:native_test_support",
+    ]
+
+    sources = [
+      "//chrome/browser/android/customtabs/custom_tabs_browsertest.cc",
```

Comment:
why are we including these chrome browser tests?

Reply:
all of these targets and tests are copied from upstream which is both unnecessary and also doesn't accomplish anything for us. We need to run Brave browser tests, not upstream browser tests.

Reply:
This is only the first step of making working broswer tests. Android tests are not like desktop ones, they need special driving apks. It does not make sense to right a new one so the plan here is to make the upstream one work first, and then modify it according to needs.